crest/Assets/Crest/Crest/Shaders/OceanInputs/Resources/AnimWavesGerstnerGeometry.shader

```diff
@@ -0,0 +1,158 @@
+		Cull Off
+
+        Pass
+        {
+            CGPROGRAM
+            #pragma vertex vert
+            #pragma fragment frag
+			#pragma enable_d3d11_debug_symbols
+
+            #include "UnityCG.cginc"
+
+			#include "../../OceanGlobals.hlsl"
+			#include "../../OceanInputsDriven.hlsl"
+			#include "../../OceanHelpersNew.hlsl"
+
+			struct appdata
+            {
+                float4 vertex : POSITION;
+                float2 axis : TEXCOORD0;
+				float2 distToSplineEnd_invNormDistToShoreline : TEXCOORD1;
+            };
+
+            struct v2f
+            {
+				float4 vertex : SV_POSITION;
+				float3 uv_slice : TEXCOORD1;
+				float2 axis : TEXCOORD2;
+				float3 worldPosScaled : TEXCOORD3;
+				float2 distToSplineEnd_invNormDistToShoreline : TEXCOORD4;
+            };
+
+			Texture2DArray _WaveBuffer;
+
+			CBUFFER_START(GerstnerPerMaterial)
+			half _FeatherWaveStart;
+			half _FeatherFromSplineEnds;
+			float _RespectShallowWaterAttenuation;
+			CBUFFER_END
+
+			CBUFFER_START(CrestPerOceanInput)
+			int _WaveBufferSliceIndex;
+			float _AverageWavelength;
+			float _AttenuationInShallows;
+			float _Weight;
+			CBUFFER_END
+
+            v2f vert(appdata v)
+            {
+                v2f o;
+
+				const float3 positionOS = v.vertex.xyz;
+				o.vertex = UnityObjectToClipPos(positionOS);
+				const float3 worldPos = mul( unity_ObjectToWorld, float4(positionOS, 1.0) ).xyz;
+
+				// UV coordinate into the cascade we are rendering into
+				o.uv_slice.xyz = WorldToUV(worldPos.xz, _CrestCascadeData[_LD_SliceIndex], _LD_SliceIndex);
+
+				// World pos prescaled by wave buffer size, suitable for using as UVs in fragment shader
+				const float waveBufferSize = 0.5f * (1 << _WaveBufferSliceIndex);
+				o.worldPosScaled = worldPos / waveBufferSize;
+
+				o.distToSplineEnd_invNormDistToShoreline = v.distToSplineEnd_invNormDistToShoreline;
+
+				// Rotate forward axis around y-axis into world space
+				o.axis = v.axis.x * unity_ObjectToWorld._m00_m20 + v.axis.y * unity_ObjectToWorld._m02_m22;
+
+                return o;
+            }
+
+            float4 frag(v2f input) : SV_Target
+            {
+				float wt = _Weight;
+
+				// Attenuate if depth is less than half of the average wavelength
+				const half depth = _LD_TexArray_SeaFloorDepth.SampleLevel(LODData_linear_clamp_sampler, input.uv_slice.xyz, 0.0).x;
+				const half depth_wt = saturate(2.0 * depth / _AverageWavelength);
+				const float attenuationAmount = _AttenuationInShallows * _RespectShallowWaterAttenuation;
+				wt *= attenuationAmount * depth_wt + (1.0 - attenuationAmount);
+
+				// Feature at front/back
+				wt *= min( input.distToSplineEnd_invNormDistToShoreline.y / _FeatherWaveStart, 1.0 );
+				if( _FeatherFromSplineEnds > 0.0 ) wt *= saturate( input.distToSplineEnd_invNormDistToShoreline.x / _FeatherFromSplineEnds );
+
+				// Quantize wave direction and interpolate waves
+				float axisHeading = atan2( input.axis.y, input.axis.x ) + 2.0 * 3.141592654;
```

Comment:
If I negate input.axis here, the waves move away from the spline rather than always towards it. Do you think this would be an option users will want?

Reply:
the wind direction setting on the ShapeGerstner component was intended to set the direction, relative to the local spline direction. however testing now it seems like its broken. do you think i should fix this up, or do you have thoughts how best to achieve this?

Reply:
I grabbed the code from AnimWavesGerstner and it works well enough using wind (further up in the vertex part):
```shader
o.axis = dot(v.axis, _AxisX) * unity_ObjectToWorld._m00_m20 + dot(v.axis, float2(-_AxisX.y, _AxisX.x)) * unity_ObjectToWorld._m02_m22;
```

Wind is kind of a weird name for this, but wind seems to just be a generic direction property anyway so I think go ahead with it.

Reply:
oh yeah that works. ive done that now.

i proposed renaming the 'wind' setting to *Wave Direction Heading Angle*. bit of a mouthful but good to be explicit i guess, if it makes sense to you? lmk

Reply:
Sounds good